prompt_toolkit/output/color_depth.py

```diff
@@ -62,7 +69,8 @@ def default(cls, term: Optional[str] = None) -> "ColorDepth":
         # vt100 escape sequences with ENABLE_VIRTUAL_TERMINAL_PROCESSING are
         # supported. We don't have a reliable way yet to know whether our
         # console supports true color or only 4-bit.
-        if is_windows() and "PROMPT_TOOLKIT_COLOR_DEPTH" not in os.environ:
+        windows = output.is_windows() if output is not None else is_windows()
+        if windows and "PROMPT_TOOLKIT_COLOR_DEPTH" not in os.environ:
```

Comment:
I think here we probably don't need a specialization for Windows, but instead, the Output classes can have a method
``get_default_color_depth()`` which is called here.

Reply:
Right, that makes sense!